modules/contact/src/constraints/ExplicitDynamicsContactConstraint.C

```diff
@@ -318,14 +287,18 @@ ExplicitDynamicsContactConstraint::solveImpactEquations(const Node & node,
   auto & u_dot = *_sys.solutionUDot();
 
   // Get lumped mass value
-  auto & diag = _sys.getVector("mass_matrix_diag");
-  Real mass_q = diag(dof_x);
+  const auto & diag = _sys.getVector("mass_matrix_diag_inverted");
+
+  Real mass_node = 1.0 / diag(dof_x);
+  Real mass_face = computeFaceMass(diag);
+
+  Real mass_eff = (mass_face * mass_node) / (mass_face + mass_node);
```

Comment:
Do we or should we assert for non-zero masses anywhere?

Reply:
For explicit dynamics, we would want an assert if any of the lumped entries are zero (not just in contact). For a zero mass we would have an infinite acceleration. I can do that in another PR because it's not reliant on contact at all.